lib/api/readable.js

```diff
@@ -146,15 +146,41 @@ module.exports = class BodyReadable extends Readable {
 
   async dump (opts) {
     let limit = opts && Number.isFinite(opts.limit) ? opts.limit : 262144
+    const signal = opts && opts.signal
+    if (signal) {
+      if (typeof signal !== 'object' || !('aborted' in signal)) {
+        throw new InvalidArgumentError('signal must be an AbortSignal')
+      }
+      if (!signal.throwIfAborted) {
+        // add in a throwIfAborted method to the signal
+        signal.throwIfAborted = function () {
```

Comment:
I don't think adding a public method here is a good solution, it'd be better to add it as a utility functions

```js
function throwIfAborted (signal) {
  if (signal.throwIfAborted) {
    signal.throwIfAborted()
  } else {
    // ...
  }
}
```

Reply:
Updating

Reply:
Updated, does it look ok?